src/sentry/static/sentry/app/components/group/suggestedOwners/suggestedOwners.tsx

```diff
@@ -0,0 +1,194 @@
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.event && prevProps.event) {
+      if (this.props.event.id !== prevProps.event.id) {
+        //two events, with different IDs
+        this.fetchData(this.props.event);
+      }
+      return;
+    }
+
+    if (this.props.event) {
+      //going from having no event to having an event
+      this.fetchData(this.props.event);
+    }
+  }
+
+  async fetchData(event: Event) {
+    if (!event) {
+      return;
+    }
+
+    // No committers if you don't have any releases
+    if (!!this.props.group.firstRelease) {
+      this.fetchCommitters(event.id);
+    }
+
+    this.fetchOwners(event.id);
+  }
+
+  fetchCommitters = async (eventId: Event['id']) => {
+    const {api, project, organization} = this.props;
+    // TODO: move this into a store since `EventCause` makes this exact request as well
+    await api
+      .requestPromise(
+        `/projects/${organization.slug}/${project.slug}/events/${eventId}/committers/`
+      )
+      .then(data => {
+        this.setState({
+          committers: data.committers,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          committers: [],
+        });
+      });
+  };
+
+  fetchOwners = async (eventId: Event['id']) => {
+    const {api, project, organization} = this.props;
+    await api
+      .requestPromise(
+        `/projects/${organization.slug}/${project.slug}/events/${eventId}/owners/`
+      )
+      .then(data => {
+        this.setState({
+          owners: data.owners,
+          rules: data.rules,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          committers: [],
+        });
+      });
+  };
+
+  /**
+   * Combine the commiter and ownership data into a single array, merging
+   * users who are both owners based on having commits, and owners matching
+   * project ownership rules into one array.
+   *
+   * The return array will include objects of the format:
+   *
+   * {
+   *   actor: <
+   *    type,              # Either user or team
+   *    SentryTypes.User,  # API expanded user object
+   *    {email, id, name}  # Sentry user which is *not* expanded
+   *    {email, name}      # Unidentified user (from commits)
+   *    {id, name},        # Sentry team (check `type`)
+   *   >,
+   *
+   *   # One or both of commits and rules will be present
+   *
+   *   commits: [...]  # List of commits made by this owner
+   *   rules:   [...]  # Project rules matched for this owner
+   * }
+   */
+  getOwnerList() {
+    const owners = this.state.committers.map(commiter => ({
+      actor: {...commiter.author, type: 'user' as Actor['type']},
+      commits: commiter.commits,
+    })) as OwnerList;
+
+    this.state.owners.forEach(owner => {
+      const normalizedOwner = {
+        actor: owner,
+        rules: findMatchedRules(this.state.rules || [], owner),
+      };
+
+      const existingIdx = owners.findIndex(o => o.actor.email === owner.email);
+      if (existingIdx > -1) {
```

Comment:
I find owners.some(...) really handy for this kind of use-case.

Reply:
Yes I agree with you, but in this case I am preferring to leave findIndex here since I am using it to update the owners array...otherwise I would have to map the owners and add a condition in the loop in order to update the correct one.

Reply:
Oh yeah, didn't notice the update 👍